index.js

```diff
@@ -1,66 +1,190 @@
+            // Allow less strict formatting of function parentheses.
+            'function-paren-newline': 'off',
+            // Prefer destructuring for certain cases only. Forcing destructuring for arrays can make the code less readable.
+            'prefer-destructuring': ['error', {
+                VariableDeclarator: {
+                    array: false,
+                    object: true,
+                },
+                AssignmentExpression: {
+                    array: false,
+                    object: false,
+                },
+            }, {
+                enforceForRenamedProperties: false,
+            }],
+            // So that the functions calling return show up in stack traces in case of an error.
+            'no-return-await': 'off',
+            // Do not enforce newlines between class members if they have a single line.
+            'lines-between-class-members': ['error', 'always', {
+                exceptAfterSingleLine: true,
+            }],
+
+            // Rules related to eslint-plugin-import.
+            // Force external modules to be specified in the package.json.
+            'import/no-extraneous-dependencies': ['error', {
+                devDependencies: true,
+            }],
+            // Force the use of named exports.
+            'import/no-default-export': 'error',
+            // It's ok to not have a default export and we force named exports anyway.
+            'import/prefer-default-export': 'off',
+            'import/no-unresolved': 'off',
+            'import/extensions': 'off',
+            // Force ordering of imports.
+            'import/order': ['error', {
+                groups: ['builtin', 'external', ['parent', 'sibling'], 'index', 'object'],
+                alphabetize: {
+                    order: 'asc',
+                    caseInsensitive: true,
+                },
+                'newlines-between': 'always',
+            }],
+            'import/no-named-as-default': 'off',
+            'import/no-import-module-exports': 'off',
+        },
     },
-    "parserOptions": {
-        "ecmaVersion": 2022,
+    // Configuration for TypeScript files only.
+    // Use the recommended TypeScript configs as a base, follow up with our overrides.
+    ...typescriptEslint.configs.recommended.map((conf) => ({
+        ...conf,
+
+        files: ['**/*.ts', '**/*.tsx', '**/*.mts', '**/*.cts'],
+    })),
+    {
+        files: ['**/*.ts', '**/*.tsx', '**/*.mts', '**/*.cts'],
+
+        plugins: {
+            '@typescript-eslint': typescriptEslint.plugin,
+        },
+
+        languageOptions: {
+            parser: typescriptEslint.parser,
+            ecmaVersion: 2022,
+        },
+
+        rules: {
+            // TypeScript doesn't support imports with `.ts` extensions, and `.js` makes the rule complain.
+            'import/extensions': 'off',
+            // Allow semicolons at the end of the code block.
+            '@typescript-eslint/no-extra-semi': 'off',
```

Comment:
I would turn this on 🙊 Autofix would probably fix all those.

Reply:
Ok. I don't see any actual reason for it being disabled.